safedata_validator/summary.py

```diff
@@ -212,13 +99,133 @@ def __init__(self, resources: Resources):
         )
         self.external_files = None
         self.data_worksheets: list[Worksheet] = []
+        self.use_project_ids = resources.use_project_ids
 
         self._rows: dict = {}
         self._ncols: int
         self.n_errors: int = 0
         self.valid_pid: Optional[list[int]] = None
         self.validate_doi = False
 
+        # Class attribute to define the metadata that may be present.
+        # These are defined in blocks of related rows described in 4-tuples:
+        #  0: a list of tuples giving the rows in the block:
+        #     * row header key
+        #     * mandatory within the block
+        #     * 'internal' name - also maps to Zenodo fields
+        #     * accepted types
+        #     * list of aliases for the fieldname (this generally will be blank)
```

Comment:
Yup - I think this is the best way to handle this. A generic solution.

Reply:
Yeah important to bear in mind that it only works properly for mandatory core fields, but should be easily extensible. I'm guessing that is fine for now? I've added some TODO's about where it would need to be extended